### DISCUSSION TITLE
Generation from HDR10+ shot on Galaxy Note20 Ultra creates rapidly flickering video, normal?  Quirk?

### DISCUSSION THREAD

**Discussion by ghost** [quietvoid/dovi_tool#127] [Q&A]
It seems like some smoothing of the HDR10+ data must be done during playback on the device.  This is the only device I own capable of playing or generating HDR10+ but it was made by Samsung, and they're the main company that still cares so it seems like it would be creating correct data, but I can't be sure since normally you'd generate it after the whole video was shot and presumably that generator has access to more frame data and can smooth things itself.   When I generate dolby vision metadata from it the PQ values being chosen vary rapidly from frame-to-frame and I end up with a video that flickers heavily during playback.  Madmeasure  generation worked fine on the same video, of course.

I can shoot a short HDR10+ video clip for you to examine if you care about this.   I posted here instead of filing an actual bug, because I don't consider it a big deal.  This particular phone has an ungodly high full-white nit level and a corresponding wider brightness range than anything short of expensive mastering displays and I still wasn't particularly impressed by HDR10+ vs regular HDR10 video on it so take it as you will.  If you're interested in looking at it I can file a bug, otherwise I'd rather not distract you from working on FEL merging and other things...   There isn't exactly a lot of HDR10+ content out there to begin with, and I don't shoot very much video at all.  I only flipped the HDR10+ switch on to try it with dovi_tool lol.   All I've seen thus far was the 4k Wizard of Oz release, and it also had Dolby Vision so I didn't need to do anything with the HDR10+ metadata.

**Comment by quietvoid** [comment]
Not much I can say about this.
It would definitely require looking at the source sample, so that's best as an issue at `hdr10plus_tool` if the JSON shows rapidly changing metadata.

The conversion to Dolby Vision would create "shots" every time the metadata changes.
It's possible making the metadata a single shot could avoid this.

**Comment by ghost** [comment]
I did check it out.  I can't recall the names of the metadata at the
moment, but I had looked at the dovi_tool source and the attribute it was
pulling from was indeed fluctuating quite a bit;  there were 3 related
values.  Sorry I can't be more specific, I'm not looking at it at the
moment.

Samsung's HDR10+ metadata appeared to be a new shot/scene for each frame to
begin with unless that's something hdr10plus_tool started doing on
extraction (pretty sure it doesn't), so I think dovi_tool did the correct
thing in following the data.   Hard to say why it was created like that or
why it doesn't fluctuate when played on the phone.  I suspect it either
does some max-change smoothing or uses some part of the included histogram
data instead (which now that I think about it, I don't remember existing in
the Wizard of Oz 4k HDR10+, but it's been a while).

Supposedly I can trim video in-phone (and no other operations) and preserve
HDR10+ so I'll try to clip it down to the most rapidly changing section and
open an issue over there when I have time later.  Hopefully it can be
rapidly closed as invalid or wontfix and you can move to other things.  :)
Being able to record video directly as HDR10 PQ is the way bigger bonus of
the new phone features to me.

On Tue, Mar 1, 2022 at 12:13 AM quietvoid ***@***.***> wrote:

> Not much I can say about this.
> It would definitely require looking at the source sample, so that's best
> as an issue at hdr10plus_tool if the JSON shows rapidly changing metadata.
>
> The conversion to Dolby Vision would create "shots" every time the
> metadata changes.
> It's possible making the metadata a single shot could avoid this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/quietvoid/dovi_tool/discussions/127#discussioncomment-2270581>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHTZ3BFGU5TXHR4SXQX3KWLU5WRP3ANCNFSM5PS5TP4A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-Gordon

**Comment by ghost** [comment]
ST2094 App 4 is HDR10+.   I'm not sure why mediainfo displays it like that
since DoVi was another application for ST2094 IIRC but that gets detailed
info listed (maybe the lack of public docs?), along with the technicolor
variant that doesn't exist on any devices AFAIK but yeah it's confusing lol.
